src/main/java/com/denizenscript/depenizen/bukkit/events/shopkeepers/ShopKeeperTradeInitiatedScriptEvent.java

```diff
@@ -0,0 +1,62 @@
+package com.denizenscript.depenizen.bukkit.events.shopkeepers;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.depenizen.bukkit.bridges.ShopkeepersBridge;
+import com.denizenscript.depenizen.bukkit.objects.shopkeepers.ShopKeeperTag;
+import com.nisovin.shopkeepers.api.events.ShopkeeperTradeEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class ShopKeeperTradeInitiatedScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // shopkeepers player tries trading
+    //
+    // @Warning This event is called for each successful trade option a ShopKeeper offers. Canceling a trade will also cancel all successive trades that might otherwise have been triggered.
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a trade with a shopkeeper is initiated.
+    //
+    // @Context
+    // <context.recipe> Returns a ListTag(ItemTag) of the trade in the form Offered(|Offered)|Result.
```

Comment:
The list will always have 3 items I believe? just that the second item might be air (not sure if that's what you meant, but `()` generally implies it might not be there at all)